### ISSUE TITLE
az synapse role assignment create has unexpected behavior

### ISSUE THREAD

**Issue by fgheysels** [Azure/azure-cli#22078]
I have an Azure DevOps pipeline which runs on a Linux agent (`ubuntu-latest`), and this agent is currently running Azure CLI 2.35.0

In my pipeline I have a task which creates a Synapse role assignment like this:

```cli
az synapse role assignment create --workspace-name $(Synapse.Name) --role "Synapse SQL Administrator" --assignee $(FunctionApp.Name)
```

As you can see, I'm using the `--assignee` parameter.
The above statement is used to make sure that an Azure Function can connect to a database in Synapse.   However, my function fails with an exception that says that the 'login failed for user <token-identified-principal>.

When I retrieve the Synapse role assignments for my Function App using this command:

```cli
az synapse role assignment list --workspace-name $(Synapse.Name) --role "Synapse SQL Administrator" --assignee $(FunctionApp.Name)
```

I receive the following result:

```
 [
  {
    "id": "d2a8284d-f0d4-4b52-90e1-b2ae83994b72",
    "principalId": "e6cda239-1bc0-406b-b327-dd73ae2be918",
    "principalType": null,
    "roleDefinitionId": "7af0c69a-a548-47d6-aea3-d00e69bd83aa",
    "scope": "workspaces/mysynapseworkspacename"
  }
]
```

As you can see, the `principalType` property is `null`.

I have removed the Synapse Role assignment and I've executed the same CLI command again to re-create the role-assignment from my workstation (Windows, also running Azure CLI 2.35.0).
Surprisingly, when I retrieve the role-assignments for my function, I get back this result:

```
[
  {
    "id": "74d4488e-e689-4fc9-808f-dccea5b0f8eb",
    "principalId": "e6cda239-1bc0-406b-b327-dd73ae2be918",
    "principalType": "ServicePrincipal",
    "roleDefinitionId": "7af0c69a-a548-47d6-aea3-d00e69bd83aa",
    "scope": "workspaces/mysynapseworkspacename"
  }
]
```

As you can see, the `principalType` property now does have a value, and my Function App now is able to login to my Synapse database.

It looks like something goes wrong when executing the `az synapse role assignment create` command on Linux ?

I've currently worked around this by using the `az synapse role assignment create` command which takes the `--assignee-object-id` and `--assignee-principal-type` parameters, like this:

```
az synapse role assignment create --workspace-name $(Synapse.Name) --role "Synapse SQL Administrator" --assignee-object-id $(FunctionApp.Identity) --assignee-principal-type ServicePrincipal
```

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: cfc71a73-f8b7-8b89-ad44-30045b1e057d
* Version Independent ID: 10dc03f8-8337-c693-8401-a6080de80e41
* Content: [az synapse role assignment](https://docs.microsoft.com/en-us/cli/azure/synapse/role/assignment?view=azure-cli-latest#az-synapse-role-assignment-create)
* Content Source: [latest/docs-ref-autogen/synapse/role/assignment.yml](https://github.com/MicrosoftDocs/azure-docs-cli/blob/main/latest/docs-ref-autogen/synapse/role/assignment.yml)
* GitHub Login: @rloutlaw
* Microsoft Alias: **routlaw**

**Comment by yonzhan** [comment]
synapse

**Comment by SaurabhSharma-MSFT** [comment]
@fgheysels Did you try to creating a Azure Support ticket so that engineer could look into your backend logs.  Please open a [Azure Support Ticket](https://aka.ms/azsupt) so that a support engineer can check further on this issue. In case you limitations in your support plan to open support ticket please let me know.

**Comment by SaurabhSharma-MSFT** [comment]
@fgheysels We will now proceed to close this thread as I haven't heard back. If there are further questions regarding this matter, please reopen it and we will gladly continue the discussion.

**Comment by fgheysels** [comment]
I am unable to open a support ticket for this, as I no longer have access to the Azure subscription where we encountered this issue.  (I've worked around it as mentioned before).

However, I do not see what kind of information you additionally require ?  I think I've explained this and gave some details.  It comes down to the fact that creating an Azure Synapse role-assignment which uses the `--assignee` parameter is not functioning well when this is executed from an Azure DevOps hosted instance which runs Linux (I don't know if the same behavior can be seen when running on LInux):  the `principalType` property is not set, which causes the role-assignment to not 'function' correctly.
